### PR TITLE
Use credentials (username/password) from landscape keyvault

### PR DESCRIPTION
**Pull request by KimForss**
## Problem
Currently there is no way to use the same username/password for all the SIDs in an environment. Additionally there is no way to specify that a SDU have unique credentials

## Solution
Read the username/password from landscape keyvault unless otherwise specified

## Tests
<Please provide steps to test the PR>

## Notes
Followup to PR 1020

### CODE REVIEW

File: .pipeline/templates/saplandscape/create-saplandscape-steps.yml [hunk 1]
```diff
@@ -158,7 +158,7 @@ steps:
         --backend-config "container_name=tfstate" \
         --backend-config "key=${sap_landscape_key}" \
         ${repo_dir}/deploy/terraform/run/sap_landscape/
-      terraform apply -auto-approve -var-file=${input} ${repo_dir}/deploy/terraform/run/sap_landscape/
+      terraform apply -auto-approve -lock=false -var-file=${input} ${repo_dir}/deploy/terraform/run/sap_landscape/
```
Comment (YunziZhangAzure) [inline .pipeline/templates/saplandscape/create-saplandscape-steps.yml:161]: Why adding lock=false?

**Comment by YunziZhangAzure** [comment]
Please resolve conflicts and make sure the test pipeline passes or have good explanation of why it fails.

**Comment by KimForss** [comment]
Because the deployment had issues and it left the state file locked, and as I don’t have permissions to break the lease it made testing harder

________________________________

**Comment by YunziZhangAzure** [comment]
> Because the deployment had issues and it left the state file locked, and as I don’t have permissions to break the lease it made testing harder
> ________________________________ From: Yunzi Zhang <notifications@github.com> Sent: Tuesday, December 22, 2020 8:40:19 PM To: Azure/sap-hana <sap-hana@noreply.github.com> Cc: Kimmo Forss <Kimmo.Forss@microsoft.com>; Assign <assign@noreply.github.com> Subject: Re: [Azure/sap-hana] Use credentials (username/password) from landscape keyvault (#1021) @YunziZhangAzure commented on this pull request.
> ________________________________ In .pipeline/templates/saplandscape/create-saplandscape-steps.yml<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fsap-hana%2Fpull%2F1021%23discussion_r547441934&data=04%7C01%7Ckimmo.forss%40microsoft.com%7Cde0cefff014f4da3060d08d8a6a908cb%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637442592238891445%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=zVZU0CKf2l3P2N7WJ4dY%2FTAYVFW%2FXxUVGfF6R7r7Qe0%3D&reserved=0>:
> @@ -158,7 +158,7 @@ steps:
> --backend-config "container_name=tfstate" \ --backend-config "key=${sap_landscape_key}" \ ${repo_dir}/deploy/terraform/run/sap_landscape/ - terraform apply -auto-approve -var-file=${input} ${repo_dir}/deploy/terraform/run/sap_landscape/ + terraform apply -auto-approve -lock=false -var-file=${input} ${repo_dir}/deploy/terraform/run/sap_landscape/ Why adding lock=false? — You are receiving this because you were assigned. Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fsap-hana%2Fpull%2F1021%23pullrequestreview-557288435&data=04%7C01%7Ckimmo.forss%40microsoft.com%7Cde0cefff014f4da3060d08d8a6a908cb%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637442592238901439%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=1BEVVLnQzpIp%2BI8nsQFjVfl3UUfmktrOnfwPWHNSQvo%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAGCOMTDKFGYHWVB6LVBUDZTSWDRZHANCNFSM4VCQIP6Q&data=04%7C01%7Ckimmo.forss%40microsoft.com%7Cde0cefff014f4da3060d08d8a6a908cb%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637442592238901439%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=WdDOQGqo%2BXn836qEtCyzHXYSCdlawKpmL4l5IwrQ8hY%3D&reserved=0>.

The pipeline should be fixed now. If not, please let me know.

**Review by rukawata — COMMENTED** [review]
Similar comments as 1020. Thanks!

File: .pipeline/templates/saplandscape/create-saplandscape-steps.yml (hunk 1 above)
Comment (KimForss) [reply .pipeline/templates/saplandscape/create-saplandscape-steps.yml:161]: troubleshooting reasons

File: deploy/terraform/terraform-units/modules/sap_system/hdb_node/variables_local.tf [hunk 2]
```diff
@@ -202,12 +205,12 @@ locals {
...
-      os_sapadm_password     = local.os_sapadm_password,
-      xsa_admin_password     = local.xsa_admin_password,
-      cockpit_admin_password = local.cockpit_admin_password,
-      ha_cluster_password    = local.ha_cluster_password
+      db_systemdb_password   = "obsolete"
+      os_sidadm_password     = "obsolete"
+      os_sapadm_password     = "obsolete"
+      xsa_admin_password     = "obsolete"
+      cockpit_admin_password = "obsolete"
+      ha_cluster_password    = "obsolete"
```
Comment (hdamecharla) [inline deploy/terraform/terraform-units/modules/sap_system/hdb_node/variables_local.tf:213]: Hi @KimForss, I still see the creds in the output of the pipeline [link](https://dev.azure.com/azurecat-sapdeploy/sap_deployment_automation/_build/results?buildId=413&view=logs&j=ed898bca-1be4-5573-ffb5-764df6a725fe&t=79a63351-3ff3-51f2-7535-49242a09082f&l=66) Is this something we should be concerned about?

**Comment by KimForss** [comment]
These will be removed once the Ansible code is ready, so I would ignore them for now

________________________________

**Review by hdamecharla — COMMENTED** [review]
Hi @KimForss, approved #1021

**Review by hdamecharla — APPROVED** [review]
Hi @KimForss, approved #1021

**Pull request merged**